python/API_GW_EXECUTION_LOGGING_ENABLED/API_GW_EXECUTION_LOGGING_ENABLED.py

```diff
@@ -0,0 +1,365 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+    API_GW_EXECUTION_LOGGING_ENABLED
+Description:
+  Checks that methods in an Amazon API Gateway stage for deployed APIs have 'loggingLevel' as one of the values specified in the rule parameter 'loggingLevel'. The rule returns NON_COMPLIANT if any method in a stage has 'loggingLevel' set to a value not matching any of the logging levels specified in the rule parameter.
+Trigger:
+  Configuration Change on AWS::ApiGateway::Stage or AWS::ApiGatewayV2::Stage
+Reports on:
+  AWS::ApiGateway::Stage or AWS::ApiGatewayV2::Stage
+Rule Parameters:
+  loggingLevel
+  (Optional) Comma-separated list of allowed logging levels. Default is "ERROR,INFO"
+Scenarios:
+  Scenario: 1
+    Given: The rule parameter is invalid
+     Then: Return ERROR
+  Scenario: 2
+    Given: The rule parameter is valid
+       And: In the Stage configuration item, atleast one method in 'methodSettings' has the 'loggingLevel' set to a value not in rule parameter.
+     Then: Return NON_COMPLIANT
+  Scenario: 3
+    Given: The rule parameter is valid
+       And: In the Stage configuration item, all methods in 'methodSettings' have the 'loggingLevel' set to a value in rule parameter.
+     Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ApiGateway::Stage'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+DEFAULT_LOGGING_LEVEL = ["ERROR", "INFO"]
+ALLOWED_LOGGING_LEVEL_VALUES = ["ERROR", "INFO", "OFF"]

```

Comment:
I think OFF, doesn't make a lot of sense. The rule must check logging. I cannot imagine a cas where the customer want to enforce the lack of logging.